### PR TITLE
Add support of PrometheusRawMetricsProvider for the Pulsar-Proxy

### PR DESCRIPTION
**Pull request by eolivelli**

### CODE REVIEW

**Comment by eolivelli** [comment]
please don't merge using the button. I will merge the patch maintaining the original author @cbornet

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 1]
```diff
@@ -830,6 +780,100 @@ public Boolean get() {
...
+                "org.apache.pulsar.broker.web", false, attributeMap);
+        webService.addRestResources("/admin",
+                "org.apache.pulsar.broker.admin.v1", true, attributeMap);
+        webService.addRestResources("/admin/v2",
+                "org.apache.pulsar.broker.admin.v2", true, attributeMap);
+        webService.addRestResources("/admin/v3",
+                "org.apache.pulsar.broker.admin.v3", true, attributeMap);
+        webService.addRestResources("/lookup",
+                "org.apache.pulsar.broker.lookup", true, attributeMap);
+        webService.addRestResources("/topics",
```
Comment (cbornet) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:822]: This was not in the previous code

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 2]
```diff
@@ -654,71 +662,13 @@ public void start() throws PulsarServerException {
...
-                final WebSocketServlet pingPongWebSocketServlet = new WebSocketPingPongServlet(webSocketService);
-                this.webService.addServlet(WebSocketPingPongServlet.SERVLET_PATH,
-                        new ServletHolder(pingPongWebSocketServlet), true, attributeMap);
-                this.webService.addServlet(WebSocketPingPongServlet.SERVLET_PATH_V2,
-                        new ServletHolder(pingPongWebSocketServlet), true, attributeMap);
+            createMetricsServlet();
+            this.addWebServerHandlers(webService, metricsServlet, this.config);
+
+            // Refresh addresses and update configuration, since the port might have been dynamically assigned
+            if (config.getBrokerServicePort().equals(Optional.of(0))) {
```
Comment (cbornet) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:670]: This is not part of the cherry-picked PR.
Maybe this should be done in another PR by cherry-picking #12353

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java (hunk 1 above)
Comment (cbornet) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:822]: Maybe this should be done in another PR by cherry-picking #8125

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 3]
```diff
@@ -830,6 +780,100 @@ public Boolean get() {
...
+                false, attributeMap);
+
+        // Add websocket service
+        addWebSocketServiceHandler(webService, attributeMap, config);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Attempting to add static directory");
+        }
+        // Add static resources
+        webService.addStaticResources("/static", "/static");
```
Comment (cbornet) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:837]: The LOG and addStaticResources should be removed from the start() method since they are now done here

Comment (eolivelli) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:837]: removed

**Comment by eolivelli** [comment]
@cbormet I addressed your comments, PTAL again

**Comment by eolivelli** [comment]
merged
